components/script/dom/htmlformelement.rs

```diff
@@ -258,8 +262,83 @@ impl HTMLFormElementMethods for HTMLFormElement {
 
     // https://html.spec.whatwg.org/multipage/#the-form-element%3Adetermine-the-value-of-a-named-property
     fn NamedGetter(&self, name: DOMString) -> Option<RadioNodeListOrElement> {
-        // return Option::Some::<RadioNodeListOrElement>;
-        unimplemented!();
+        // unimplemented!();
+        let window = window_from_node(self);
+
+        // Q2. ERROR in the following declaration - check the attached screenshot (image2)
+        let mut candidates: Vec<Dom<HTMLElement>> = Vec::new();
```

Comment:
This will need to be DomRoot, and store Node instead of HTMLElement.

Reply:
image2:
![image2](https://user-images.githubusercontent.com/16729243/70269089-b42edc00-176f-11ea-8fb2-3f4efabed8cf.png)

Reply:
Yep, DomRoot should solve that.

Reply:
If we change this vector to contain a `DomRoot<Node>` element, should we need to change the `Dom<HTMLElement>` in `past names map` declaration or we will need to downcast the `Node` to an `HTMLElement` when inserting an entry in `past names map`?

Reply:
It should probably be enough to store `Dom<Element>` in the past names map, and that will require downcasting the node to Element, yes.

Reply:
Done.